driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -47,12 +46,8 @@ default BooleanExpression getBoolean(final String fieldName, final boolean other
 
     NumberExpression getNumber(String fieldName, NumberExpression other);
 
-    default NumberExpression getNumber(final String fieldName, final double other) {
-        return getNumber(fieldName, of(other));
-    }
-
-    default NumberExpression getNumber(final String fieldName, final Decimal128 other) {
-        return getNumber(fieldName, of(other));
+    default NumberExpression getNumber(final String fieldName, final Number other) {
+        return getNumber(fieldName, Expressions.numberToExpression(other));
     }
```

Comment:
Yes https://github.com/mongodb/mongo-java-driver/pull/1032#discussion_r1048853914 . The issue was that the one approach failed to round-trip values. But the other seemingly-consistent approach led to us having both `getInteger(int)` and `getNumber(int)` - if I as a user saw both methods, I would not see the point in having both, since we seemingly only ever need `getInteger`. So there is actually a difference (which does not apply in the case of `of`), and I have reverted to the way that seemed inconsistent.

Reply:
Discussed this over Zoom. I understand the situation and the decision now.